src/App.tsx

```diff
@@ -227,6 +228,10 @@ const Main = ({ children, ...props }: ViewProps) => {
         Alert.alert('Authentication required', 'Please authenticate to unlock your wallet.', [
           { text: 'Try again', onPress: unlockApp }
         ])
+      } else if (error.message?.includes('Could not decrypt the value with provided keychain')) {
+        console.error(e)
+        await deleteWallet()
+        navigateRootStack('LandingScreen')
```

Comment:
It seems that when uninstalling the app, SecureStore leaves some "garbage" behind, so that next time the app gets installed, we get this error.

Reply:
Couldn't that be dangerous? Are we sure that this error can only be shown because of this "garbage" left behind, and not because of some user action?

Reply:
I think I fixed the problem without this problematic fix. I set the `keychainService` option of the SecureStore `setItemAsync` (as well as `getItemAsync`) to the string `'mobile-wallet-key-service'`. This seems to not cause this exception. The way I tested it was:

- Have the app running on my phone
- Stop the server
- Change the app name
- Start the server
- I see message on my phone that says "App needs to be closed while being updated"
- After a few seconds the app launches again, it has the new name, and no error was shown (tested this with both pin and bio login, all worked 💪 )